app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -45,225 +49,86 @@
+            mMap = googleMap;
+            markerMap = googleMap;
+//            LatLng latLng = new LatLng(currentLocation.getLatitude(), currentLocation.getLongitude());
+//            MarkerOptions markerOptions = new MarkerOptions().position(latLng).title("Here I am!");
+//            googleMap.animateCamera(CameraUpdateFactory.newLatLng(latLng));
+//            googleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, 5));
+//            googleMap.addMarker(markerOptions);
+        }
+    };
 
     @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-    }
-
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        View view = inflater.inflate(R.layout.fragment_events, container, false);
-        progressDialog=new ProgressDialog(getContext());
-        progressDialog.setMessage("Fetching location...");
-        SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager()
-                .findFragmentById(R.id.map);
-        mapFragment.getMapAsync(this);
-        return view;
+    public View onCreateView( LayoutInflater inflater,
+                              ViewGroup container,
+                              Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_events, container, false);
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(getActivity());
+        fetchLastLocation();
+        SupportMapFragment mapFragment =
+                (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
+        if (mapFragment != null) {
+            mapFragment.getMapAsync(callback);
+        }
     }
 
-    @Override
-    public void onMapReady(GoogleMap googleMap) {
-        map = googleMap;
-        googleMap.addMarker(new MarkerOptions()
-                .position(new LatLng(51.5072, 0.1276))
-                .title("London"));
-        LatLng latLng = new LatLng(51.5072, 0.1276);
-        map.moveCamera(CameraUpdateFactory.newLatLng(latLng));
-        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng,15f));
+    private void placeMarkers() {
+        LatLng newLang = new LatLng(37.474300, -122.139038);
+        MarkerOptions newMarker = new MarkerOptions().position(newLang).title("Marker 1");
+        markerMap.addMarker(newMarker);
+    }
+
+    private void fetchLastLocation() {
+        if (ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
This line is WAY too long. Can we make it shorter by adding new lines strategically?